tests/chainer_tests/training_tests/extensions_tests/test_warmup_shift.py

```diff
@@ -0,0 +1,71 @@
+import unittest
+
+import mock
+
+from chainer import testing
+from chainer.training import extensions
+from chainer.training import util
+
+
+@testing.parameterize(
+    {'init': 1, 'warmup_start_lr': 0.1,
+     'warmup_iter': 100, 'expect': [0.1, 0.991, 1, 1]},
+    {'init': 0.1, 'warmup_start_lr': 1,
+     'warmup_iter': 10, 'expect': [1, 0.19, 0.1, 0.1]},
+    {'init': 1, 'warmup_start_lr': -1,
+     'warmup_iter': 10, 'expect': [-1, 0.8, 1, 1]},
+)
+class TestWarmupShift(unittest.TestCase):
+
+    def setUp(self):
+        self.optimizer = mock.MagicMock()
+        self.extension = extensions.WarmupShift(
+            'x', self.warmup_start_lr, self.init, self.optimizer)
+
+        self.interval = 1
+        self.expect = [e for e in self.expect for _ in range(self.interval)]
+        self.trigger = util.get_trigger((self.interval, 'iteration'))
+
+        self.trainer = testing.get_trainer_with_mock_updater(self.trigger)
+        self.trainer.updater.get_optimizer.return_value = self.optimizer
+
+    def _run_trainer(self, extension, expect, optimizer=None):
+        if optimizer is None:
+            optimizer = self.optimizer
+        extension.initialize(self.trainer)
+        actual = []
+        for _ in range(self.warmup_iter+2):
+            self.trainer.updater.update()
+            actual.append(optimizer.x)
+            if self.trigger(self.trainer):
+                extension(self.trainer)
+
+        self.assertEqual(round(actual[0], 6), expect[0])
```

Comment:
Please use `chainer.testing.assert_allclose` instead of manually rounding values.